File: app/src/main/java/com/karthik/splash/Models/PhotosLists/Links.java
package com.karthik.splash.Models.PhotosLists;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by karthikrk on 20/12/15.
 */
public class Links implements Parcelable {
    @SerializedName("self")
    public String self;
    @SerializedName("html")
    public String html;
    @SerializedName("photos")
    public String photos;
    @SerializedName("likes")
    public String likes;
    @SerializedName("portfolio")
    @Expose
    public String portfolio;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.self);
        dest.writeString(this.html);
        dest.writeString(this.photos);
        dest.writeString(this.likes);
        dest.writeString(this.portfolio);
    }

    public Links() {
    }

    protected Links(Parcel in) {
        this.self = in.readString();
        this.html = in.readString();
        this.photos = in.readString();
        this.likes = in.readString();
        this.portfolio = in.readString();
    }

    public static final Parcelable.Creator<Links> CREATOR = new Parcelable.Creator<Links>() {
        @Override
        public Links createFromParcel(Parcel source) {
            return new Links(source);
        }

        @Override
        public Links[] newArray(int size) {
            return new Links[size];
        }
    };
}

File: app/src/main/java/com/karthik/splash/models/photoslists/Links.kt
package com.karthik.splash.models.photoslists

import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

/**
 * Created by karthikrk on 20/12/15.
 */
@Parcelize
data class Links(
        @SerializedName("self")
        var self: String? = null,
        @SerializedName("html")
        var html: String? = null,
        @SerializedName("photos")
        var photos: String? = null,
        @SerializedName("likes")
        var likes: String? = null,
        @SerializedName("portfolio")
        @Expose
        var portfolio: String? = null
) : Parcelable
